### PR TITLE
Feature/dotnet 6 upgrade

### PR DESCRIPTION
**Pull request by aaronbloom**
- Upgrade to .NET6
- Move to centrally managed package versions
- Move most specific project config into `Directory.Build.props`

Latest VCEL.Benchmark results
```
BenchmarkDotNet v0.13.8, Windows 10 (10.0.19045.3324/22H2/2022Update)
Intel Core i7-10700KF CPU 3.80GHz, 1 CPU, 16 logical and 8 physical cores
.NET SDK 7.0.401
  [Host]   : .NET 6.0.22 (6.0.2223.42425), X64 RyuJIT AVX2
  .NET 6.0 : .NET 6.0.22 (6.0.2223.42425), X64 RyuJIT AVX2

Job=.NET 6.0  Runtime=.NET 6.0  

```
| Method                  | Mean       | Error     | StdDev    | Ratio | RatioSD | Gen0   | Allocated | Alloc Ratio |
|------------------------ |-----------:|----------:|----------:|------:|--------:|-------:|----------:|------------:|
| NullEquals              |  0.6800 ns | 0.0305 ns | 0.0285 ns |  0.09 |    0.01 |      - |         - |        0.00 |
| NullEqualsChecked       |  0.6903 ns | 0.0376 ns | 0.0432 ns |  0.09 |    0.01 |      - |         - |        0.00 |
| ShortEquals             |  8.4164 ns | 0.1924 ns | 0.2633 ns |  1.09 |    0.06 | 0.0057 |      48 B |        1.00 |
| ShortEqualsChecked      |  8.2534 ns | 0.1860 ns | 0.2285 ns |  1.07 |    0.05 | 0.0057 |      48 B |        1.00 |
| UshortEquals            |  8.4942 ns | 0.1939 ns | 0.2589 ns |  1.10 |    0.06 | 0.0057 |      48 B |        1.00 |
| UshortEqualsChecked     |  8.7195 ns | 0.2008 ns | 0.4057 ns |  1.15 |    0.08 | 0.0057 |      48 B |        1.00 |
| IntEquals               |  7.6273 ns | 0.1791 ns | 0.3184 ns |  1.00 |    0.00 | 0.0057 |      48 B |        1.00 |
| IntEqualsChecked        |  8.1435 ns | 0.1862 ns | 0.2486 ns |  1.06 |    0.05 | 0.0057 |      48 B |        1.00 |
| UintEquals              |  7.7852 ns | 0.1806 ns | 0.2703 ns |  1.02 |    0.05 | 0.0057 |      48 B |        1.00 |
| UintEqualsChecked       |  7.9110 ns | 0.1838 ns | 0.2117 ns |  1.03 |    0.06 | 0.0057 |      48 B |        1.00 |
| LongEquals              |  8.6871 ns | 0.1998 ns | 0.3653 ns |  1.14 |    0.08 | 0.0057 |      48 B |        1.00 |
| LongEqualsChecked       |  9.6425 ns | 0.1865 ns | 0.2291 ns |  1.25 |    0.06 | 0.0057 |      48 B |        1.00 |
| UlongEquals             |  9.6776 ns | 0.2107 ns | 0.5362 ns |  1.28 |    0.08 | 0.0057 |      48 B |        1.00 |
| UlongEqualsChecked      |  8.4670 ns | 0.1952 ns | 0.4243 ns |  1.12 |    0.06 | 0.0057 |      48 B |        1.00 |
| DoubleEquals            |  8.1269 ns | 0.1870 ns | 0.2912 ns |  1.06 |    0.05 | 0.0057 |      48 B |        1.00 |
| DoubleEqualsChecked     |  7.9656 ns | 0.1797 ns | 0.2207 ns |  1.03 |    0.06 | 0.0057 |      48 B |        1.00 |
| DecimalEquals           | 11.6379 ns | 0.2499 ns | 0.3504 ns |  1.52 |    0.08 | 0.0076 |      64 B |        1.33 |
| DecimalEqualsChecked    | 11.8460 ns | 0.2455 ns | 0.4102 ns |  1.55 |    0.09 | 0.0076 |      64 B |        1.33 |
| StringEquals            |  0.9020 ns | 0.0275 ns | 0.0244 ns |  0.12 |    0.01 |      - |         - |        0.00 |
| StringEqualsChecked     |  0.9095 ns | 0.0287 ns | 0.0254 ns |  0.12 |    0.01 |      - |         - |        0.00 |
| DecimalIntEquals        | 15.4510 ns | 0.3018 ns | 0.4958 ns |  2.03 |    0.12 | 0.0067 |      56 B |        1.17 |
| DecimalIntEqualsChecked | 14.1433 ns | 0.2997 ns | 0.4102 ns |  1.84 |    0.10 | 0.0067 |      56 B |        1.17 |
| IntDecimalEquals        | 14.1214 ns | 0.3078 ns | 0.4606 ns |  1.84 |    0.10 | 0.0067 |      56 B |        1.17 |
| IntDecimalEqualsChecked | 14.0786 ns | 0.3060 ns | 0.4853 ns |  1.84 |    0.10 | 0.0067 |      56 B |        1.17 |
| ULongFloatEquals        | 14.0572 ns | 0.3030 ns | 0.4893 ns |  1.84 |    0.09 | 0.0057 |      48 B |        1.00 |
| ULongFloatEqualsChecked | 13.9996 ns | 0.3012 ns | 0.4600 ns |  1.83 |    0.12 | 0.0057 |      48 B |        1.00 |
| FloatUlongEquals        | 14.1607 ns | 0.3085 ns | 0.3901 ns |  1.84 |    0.09 | 0.0057 |      48 B |        1.00 |
| FloatUlongEqualsChecked | 14.1156 ns | 0.3032 ns | 0.4251 ns |  1.84 |    0.10 | 0.0057 |      48 B |        1.00 |
```

### CODE REVIEW

File: src/Directory.Packages.props [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    <ManagePackageVersionsCentrally>true</ManagePackageVersionsCentrally>
+  </PropertyGroup>
+  <ItemGroup>
+    <PackageVersion Include="Antlr4" Version="4.6.6" />
+    <PackageVersion Include="Antlr4.Runtime" Version="4.6.6" />
+    <PackageVersion Include="BenchmarkDotNet" Version="0.13.8" />
+    <PackageVersion Include="Microsoft.CSharp" Version="4.7.0" />
+    <PackageVersion Include="Microsoft.CodeAnalysis.CSharp" Version="4.7.0" />
+    <PackageVersion Include="Microsoft.CodeAnalysis.Common" Version="4.7.0" />
+    <PackageVersion Include="Microsoft.NET.Test.Sdk" Version="17.7.2" />
```
Comment (bgever) [inline src/Directory.Packages.props:12]: .NET 6 SDK has the Test Tool (dotnet test) v17.3.3. Wondering whether this doesn't work well with this package?

File: src/Tests/VCEL.Test/FunctionTests.cs [hunk 2]
```diff
@@ -186,13 +186,13 @@ public void EvalToday()
             }
         }
 
-        [TestCase("lowercase('ABCD')", "abcd")]
-        [TestCase("uppercase('abcd')", "ABCD")]
+        // [TestCase("lowercase('ABCD')", "abcd")]
```
Comment (bgever) [inline src/Tests/VCEL.Test/FunctionTests.cs:189]: Remove or keep?

File: src/VCEL.CSharp/Expression/Func/DefaultCSharpFunctions.cs [hunk 3]
```diff
@@ -63,21 +63,27 @@ public DefaultCSharpFunctions()
...
+                    var startIndex = args[1]?.ToString() is { } startString && int.TryParse(startString, out var start)
+                        ? start
+                        : throw new ArgumentException("Invalid start index");
+                    var strLength = args[2]?.ToString() is { } lengthString && int.TryParse(lengthString, out var length)
+                        ? length
+                        : throw new ArgumentException("Invalid length");
+                    return $"{sourStr}.Substring({startIndex}, {strLength})";
+                }
+                default:
+                    return null;
```
Comment (bgever) [inline src/VCEL.CSharp/Expression/Func/DefaultCSharpFunctions.cs:85]: Throw error instead?

File: src/VCEL.CSharp/Expression/Func/DefaultCSharpFunctions.cs [hunk 4]
```diff
@@ -63,21 +63,27 @@ public DefaultCSharpFunctions()
             switch (args.Length)
             {
                 case 2:
-                    {
-                        var sourStr = args[0]?.ToString();
-                        var startIndex = int.Parse(args[1]?.ToString());
-                        return $"{sourStr}.Substring({startIndex})";
-                    }
+                {
+                    var sourStr = args[0]?.ToString();
```
Comment (bgever) [inline src/VCEL.CSharp/Expression/Func/DefaultCSharpFunctions.cs:67]: I assume it meant to be source?
```suggestion
                    var sourceString = args[0]?.ToString();
```

**Review by bgever — APPROVED** [review]
Added some minor remarks.

File: src/VCEL.CSharp/Expression/Func/DefaultCSharpFunctions.cs (hunk 3 above)
Comment (aaronbloom) [reply src/VCEL.CSharp/Expression/Func/DefaultCSharpFunctions.cs:85]: I did not want to change the existing logic as part of these changes

File: src/Directory.Packages.props (hunk 1 above)
Comment (aaronbloom) [reply src/Directory.Packages.props:12]: Build and test passes, so should be fine

**Review by brifly — APPROVED** [review]

**Pull request merged**